test/OpenTelemetry.Tests/Shared/TagTransformerJsonHelperTest.cs

```diff
@@ -0,0 +1,154 @@
+        }
+
+        [Theory]
+        [InlineData(new object[] { new short[] { } })]
+        [InlineData(new object[] { new short[] { 0 } })]
+        [InlineData(new object[] { new short[] { short.MaxValue, short.MinValue, 4, 13 } })]
+        public void ShortArray(short[] data)
+        {
+            VerifySerialization(data);
+        }
+
+        [Theory]
+        [InlineData(new object[] { new ushort[] { } })]
+        [InlineData(new object[] { new ushort[] { 0 } })]
+        [InlineData(new object[] { new ushort[] { ushort.MaxValue, ushort.MinValue, 4, 13 } })]
+        public void UShortArray(ushort[] data)
+        {
+            VerifySerialization(data);
+        }
+
+        [Theory]
+        [InlineData(new object[] { new int[] { } })]
+        [InlineData(new object[] { new int[] { 0 } })]
+        [InlineData(new object[] { new int[] { int.MaxValue, int.MinValue, 4, 13 } })]
+        public void IntArray(int[] data)
+        {
+            VerifySerialization(data);
+        }
+
+        [Theory]
+        [InlineData(new object[] { new uint[] { } })]
+        [InlineData(new object[] { new uint[] { 0 } })]
+        [InlineData(new object[] { new uint[] { uint.MaxValue, uint.MinValue, 4, 13 } })]
+        public void UIntArray(uint[] data)
+        {
+            VerifySerialization(data);
+        }
+
+        [Theory]
+        [InlineData(new object[] { new long[] { } })]
+        [InlineData(new object[] { new long[] { 0 } })]
+        [InlineData(new object[] { new long[] { long.MaxValue, long.MinValue, 4, 13 } })]
+        public void LongArray(long[] data)
+        {
+            VerifySerialization(data);
+        }
+
+        [Theory]
+        [InlineData(new object[] { new ulong[] { } })]
+        [InlineData(new object[] { new ulong[] { 0 } })]
+        [InlineData(new object[] { new ulong[] { ulong.MaxValue, ulong.MinValue, 4, 13 } })]
+        public void ULongArray(ulong[] data)
+        {
+            VerifySerialization(data);
+        }
+
+        [Theory]
+        [InlineData(new object[] { new float[] { } })]
+        [InlineData(new object[] { new float[] { 0 } })]
+        [InlineData(new object[] { new float[] { float.MaxValue, float.MinValue, 4, 13 } })]
+        [InlineData(new object[] { new float[] { float.Epsilon } })]
+        public void FloatArray(float[] data)
+        {
+            VerifySerialization(data);
+        }
+
+        [Theory]
+        [InlineData(new object[] { new double[] { } })]
+        [InlineData(new object[] { new double[] { 0 } })]
+        [InlineData(new object[] { new double[] { double.MaxValue, double.MinValue, 4, 13 } })]
+        [InlineData(new object[] { new double[] { double.Epsilon } })]
+        public void DoubleArray(double[] data)
+        {
+            VerifySerialization(data);
+        }
+
+        private static void VerifySerialization(Array data)
+        {
+            var reflectionBasedResult = System.Text.Json.JsonSerializer.Serialize(data);
```

Comment:
nit: All of these tests are declared with `new object[]` in the `InlineData` if you moved that here you might be able to simply all of those? Eg:

```csharp
var testData = new object[] { data };
var reflectionBasedResult = System.Text.Json.JsonSerializer.Serialize(testData);
var rawResult = TagTransformerJsonHelper.JsonSerializeArrayTag(testData);
```

Reply:
I hit this issue for the string array : (
https://github.com/dotnet/csharpstandard/issues/268#issuecomment-647152672

Reply:
Thanks for finding the Roslyn issue - I think I started with the string test and then simply copied the pattern to all the others (even though some of them might not need it).